app/src/main/java/com/kms/cura/view/adapter/RatingListAdapter.java

```diff
@@ -0,0 +1,120 @@
+package com.kms.cura.view.adapter;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.RatingBar;
+import android.widget.TextView;
+
+import com.kms.cura.R;
+import com.kms.cura.entity.AppointmentEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by toanbnguyen on 8/25/2016.
+ */
+public class RatingListAdapter extends BaseAdapter {
+
+    private Context context;
+    private int resource;
+    private List<AppointmentEntity> appointmentEntities;
+    private ArrayList<Boolean> expand;
+    private boolean firstLoad;
+
+    public RatingListAdapter(Context context, int resource, List<AppointmentEntity> appointmentAll) {
+        this.context = context;
+        this.resource = resource;
+        appointmentEntities = new ArrayList<>();
+        for (AppointmentEntity entity : appointmentAll) {
+            if (entity.getRate() != 0) {
+                appointmentEntities.add(entity);
+            }
+        }
+        expand = new ArrayList<>();
+        for (AppointmentEntity entity : appointmentEntities) {
+            if (entity.getRate() != 0) {
+                expand.add(false);
```

Comment:
Why don't you do this on the upper for too, so there would be no need for this second for loop.